packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx

```diff
@@ -4,59 +4,67 @@ import { lock } from '@wordpress/icons';
 import clsx from 'clsx';
 import { useTranslate } from 'i18n-calypso';
 
-interface DomainsTableSSLCellProps {
+interface DomainsTableSslCellProps {
 	domainManagementLink: string;
 	sslStatus: DomainData[ 'ssl_status' ];
+	hasWpcomManagedSslCert?: DomainData[ 'has_wpcom_managed_ssl_cert' ];
 }
 
-export default function DomainsTableSSLCell( {
+export default function DomainsTableSslCell( {
 	domainManagementLink,
 	sslStatus,
-}: DomainsTableSSLCellProps ) {
+	hasWpcomManagedSslCert,
+}: DomainsTableSslCellProps ) {
 	const translate = useTranslate();
+	// WordPress.com managed subdomains (e.g. *.wordpress.com, *.wpcomstaging.com, etc.)
+	// are covered by a wildcard SSL cert, so we display them as 'Active'.
+	const isActiveSsl = sslStatus === 'active' || hasWpcomManagedSslCert;
+	const isPendingSsl = sslStatus === 'pending' || sslStatus === 'newly_registered';
+	const domainHasSsl = sslStatus !== null || hasWpcomManagedSslCert;
 
-	const isPendingSSL = sslStatus === 'pending' || sslStatus === 'newly_registered';
-
-	const getSSLStatusText = () => {
-		if ( sslStatus === 'active' ) {
+	const getSslStatusText = () => {
+		if ( isActiveSsl ) {
 			return translate( 'Active' );
 		}
-		if ( isPendingSSL ) {
+		if ( isPendingSsl ) {
 			return translate( 'Pending' );
 		}
 		if ( sslStatus === 'disabled' ) {
 			return translate( 'Disabled' );
 		}
 	};
 
+	const getSslStatusClassnames = ( element: 'button' | 'icon' ) =>
+		clsx( `domains-table-row__ssl-status-${ element }`, {
+			[ `domains-table-row__ssl-status-${ element }__active` ]: isActiveSsl,
+			[ `domains-table-row__ssl-status-${ element }__pending` ]: isPendingSsl,
+			[ `domains-table-row__ssl-status-${ element }__disabled` ]: sslStatus === 'disabled',
+		} );
```

Comment:
Nit, but I would argue that verbosity is preferable here because it makes it easier to grep for those class names across the code base (i.e., if I search for `domains-table-row__ssl-status-button` across the code base, I won't find this instance).

However, I know we have several instances in `packages/domains-table/src/domains-table/style.scss` of class names being stringed together in a "similar" way (e.g., [here](https://github.com/Automattic/wp-calypso/blob/c31789c95098eaffebc05b8edfb417c64580d99e/packages/domains-table/src/domains-table/style.scss#L152)). So, there's no rule against this, but I would personally argue that it's preferable to spell out the full class name.

Reply:
A good point, thanks. Your next question on moving the modifier classes to the parent `td` helps a lot too, alleviating the need for a second set of modifiers.